### PR TITLE
fix bcdb masterslave and threebotserver start

### PR DESCRIPTION
**Pull request by waleedhammam**
- Still needs review and cleanup

### CODE REVIEW

File: JumpscaleCore/data/bcdb/BCDB.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 
 class BCDB(j.baseclasses.object):
-    def _init(self, name=None, storclient=None, reset=False, readonly=False):
+    def _init(self, name=None, storclient=None, reset=False, readonly=True):
```
Comment (rkhamis) [inline JumpscaleCore/data/bcdb/BCDB.py]: we don't need this anymore

File: JumpscaleCore/data/bcdb/BCDBFactory.py [hunk 2]
```diff
@@ -36,41 +36,50 @@ def _init(self, **kwargs):
...
-                    timeout = j.data.time.epoch + 15
-                    while j.data.time.epoch < timeout:
-                        if j.core.db.get("threebot.starting") is None:
-                            self.__master = False
-                            return (
-                                self.__master
-                            )  # means we found a threebot who was started properly, can now start as slave
-                raise j.exceptions.Base("threebotserver is starting but did not succeed within 60+15 sec")
-
+            #     # see if a threebot starting
```
Comment (rkhamis) [inline JumpscaleCore/data/bcdb/BCDBFactory.py]: cleanup

File: JumpscaleCore/servers/myjobs/MyJobsFactory.py [hunk 3]
```diff
@@ -618,6 +619,10 @@ def wait_queue(self, queue_name, size=1, timeout=120, die=True):
         if res:
             return res[0]
 
+    def _acquire_bcdb_master_(self):
+        j.data.bcdb._master_set(True)
+        j.clients.bcdbmodel._init()
```
Comment (rkhamis) [inline JumpscaleCore/servers/myjobs/MyJobsFactory.py]: let's create a reset method

**Review by rkhamis — CHANGES_REQUESTED** [review]

File: JumpscaleCore/data/bcdb/BCDBFactory.py (hunk 2 above)
Comment (waleedhammam) [reply JumpscaleCore/data/bcdb/BCDBFactory.py]: cleaned

File: JumpscaleCore/data/bcdb/BCDB.py (hunk 1 above)
Comment (waleedhammam) [reply JumpscaleCore/data/bcdb/BCDB.py]: done

File: JumpscaleCore/servers/myjobs/MyJobsFactory.py (hunk 3 above)
Comment (waleedhammam) [reply JumpscaleCore/servers/myjobs/MyJobsFactory.py]: done and removed

File: JumpscaleCore/servers/threebot/ThreeBotServersFactory.py [hunk 4]
```diff
@@ -28,18 +28,19 @@ def _threebot_starting(self, starting=True):
         print("MARK THREEBOT IS STARTING")
 
         j.threebot.active = True
-        if j.core.db and starting:
-            j.core.db.set("threebot.starting", ex=120, value="1")
+        # if j.core.db and starting:
+        #     j.core.db.set("threebot.starting", ex=120, value="1")
```
Comment (waleedhammam) [inline JumpscaleCore/servers/threebot/ThreeBotServersFactory.py]: @rkhamis shall we cleanup this too ?

File: JumpscaleCore/servers/threebot/ThreeBotServersFactory.py [hunk 5]
```diff
@@ -28,18 +28,19 @@ def _threebot_starting(self, starting=True):
...
-            return True
-        res = j.sal.nettools.tcpPortConnectionTest("localhost", 6380, timeout=0.1)
-        return res
+    # FIXME: NOT USED ANYWHERE IS IT REQUIRED ?
+    # def threebotserver_check(self):
+    #     if j.core.db and j.core.db.get("threebot.starting"):
+    #         self.threebotserver_require()
+    #         return True
+    #     res = j.sal.nettools.tcpPortConnectionTest("localhost", 6380, timeout=0.1)
+    #     return res
```
Comment (waleedhammam) [inline JumpscaleCore/servers/threebot/ThreeBotServersFactory.py]: @rkhamis also shall we cleanup this too ?

Comment (rkhamis) [reply JumpscaleCore/servers/threebot/ThreeBotServersFactory.py]: yes

File: JumpscaleCore/servers/threebot/ThreeBotServersFactory.py (hunk 4 above)
Comment (rkhamis) [reply JumpscaleCore/servers/threebot/ThreeBotServersFactory.py]: yup! clean code is good code

**Review by rkhamis — CHANGES_REQUESTED** [review]

**Review by rkhamis — APPROVED** [review]

File: JumpscaleCore/data/bcdb/BCDB.py [hunk 6]
```diff
@@ -50,7 +50,7 @@ def _init(self, name=None, storclient=None, reset=False, readonly=False):
         self.storclient = storclient
 
         j.sal.fs.createDir(self._data_dir)
-        self.readonly = readonly
+        self.readonly = not j.data.bcdb._master
```
Comment (xmonader) [inline JumpscaleCore/data/bcdb/BCDB.py]: let's remove readonly from constructor

Comment (rkhamis) [reply JumpscaleCore/data/bcdb/BCDB.py]: it changes the function signature, I really don't want us to start refactoring at this point

**Review by rkhamis — APPROVED** [review]

**Pull request merged**